components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/introspection/OAuth2IntrospectionEndpoint.java

```diff
@@ -84,10 +84,10 @@ public Response introspect(@FormParam("token") String token, @FormParam("token_t
                     entity("{\"error\": \"" + INVALID_INPUT + "\"}").build();
         }
 
-        String[] claimsUris;
+        String[] claimsUris = null;
         if (StringUtils.isNotEmpty(requiredClaims)) {
             claimsUris = requiredClaims.split(",");
-        } else {
+        } else if (requiredClaims != null && requiredClaims.length() == 0) {
```

Comment:
What is the reason of having requiredClaims.length() == 0?

Reply:
Sending empty value in required_claims should return a JWT with no custom claims within the introspection response according to the description here (wso2/docs-is#1373). In order to find whether the required_claims param is sent with the request, we use this `claimsUris` string array (If it is null, required_claims param is not sent, if it is not null, required_claims param is sent either with a value or without a value) . As we need send a JWT when required_claims is an empty value, we need to set `claimsUris` array to an empty array.

Reply:
Shall we use StringUtils.isNotBlank(requiredClaims)?

Reply:
Here we need to check both the conditions `requiredClaims != null` and `requiredClaims.length() == 0`. If we get `requiredClaims=""`, this check should return `true`. But StringUtils.isNotBlank("")  = false. Therefore, we cannot use StringUtils.isNotBlank(requiredClaims) here.